synfig-studio/src/gui/workarea.cpp

```diff
@@ -1536,11 +1545,8 @@ WorkArea::on_drawing_area_event(GdkEvent *event)
 			double y(event->button.y), x(event->button.x);
 
 			// Erase the guides if dragged into the rulers
-			if(curr_guide_is_x && !std::isnan(x) && x<0.0 )
-				get_guide_list_x().erase(curr_guide);
-			else
-			if(!curr_guide_is_x && !std::isnan(y) && y<0.0 )
-				get_guide_list_y().erase(curr_guide);
+			if(!std::isnan(x) && (x<0.0 || y<0.0))
+				get_guide_list().erase(curr_guide);
```

Comment:
I think if user is rotating guide, it shouldn't be deleted if user release mouse button over a ruler.
However, I tested on Inkscape and it would be deleted there too.

Reply:
I think it also makes sense to not delete while rotating but I guess it just come down to user preferences. As you said inkscape deletes in that situation.

P.s. note to self I forgot to add here the check for if y is nan